src/common/snippets/include/snippets/lowered/pass/brgemm_blocking.hpp

```diff
@@ -32,7 +32,16 @@ class BrgemmBlockingBase {
      * @param brgemm_expr Brgemm expression
      * @return tuple in format (m_block, n_block, k_block)
      */
-    virtual std::tuple<size_t, size_t, size_t> get_blocking_params(const ov::snippets::lowered::ExpressionPtr& brgemm_expr);
+    virtual std::tuple<size_t, size_t, size_t> get_blocking_params(const ov::snippets::lowered::ExpressionPtr& brgemm_expr) const;
+    /**
+     * @interface get_blocking_params
+     * @brief Computes optimal blocking params for current dimensions M,N,K
+     * @param M dimension M
+     * @param N dimension N
+     * @param K dimension K
+     * @return tuple in format (m_block, n_block, k_block)
+     */
+    virtual std::tuple<size_t, size_t, size_t> get_blocking_params(size_t M, size_t N, size_t K) const = 0;
```

Comment:
I have the following concern: now we have 2 virtual methods which differ only by parameters, and one is called in another. From my perspective, this is unnecessary complication.
I guess it is done in order to reuse the code which extracts MNK dims from expression descriptors. If I am right, I would suggest moving this logic in a separate static method and leave only one `get_blocking_params(const ov::snippets::lowered::ExpressionPtr& brgemm_expr)` virtual method.
If I am wrong, please explain why is this needed.

Reply:
This is really great idea. Thanks a lot for the suggestion!
Done!